kclvm/compiler_base/error/src/diagnostic/pendant.rs

```diff
@@ -0,0 +1,114 @@
+//! 'pendant.rs' defines all pendants that builtin in compiler_base_error.
+use super::{style::DiagnosticStyle, Formatter};
+use rustc_errors::styled_buffer::StyledBuffer;
+
+/// LabelPendant: A pendant to shown some label messages for diagnostics.
+///
+/// e.g.
+/// error: this is an error!
+/// warning[W0011]: this is an warning!
+/// note: this is note.
+///
+/// 'error', 'warning[W0011]' and 'note' are 'LabelPendant'.
+///
+/// And `LabelPendant` currently supported text format rendering style:
+///
+/// - "error" => DiagnosticStyle::NeedFix,
+/// - "warning" => DiagnosticStyle::NeedAttention,
+/// - "help" => DiagnosticStyle::Helpful,
+/// - "note" => DiagnosticStyle::Important,
+/// - other  => DiagnosticStyle::NoStyle,
+pub struct LabelPendant {
+    /// It's just an icon for the language. e.g. Rust、Java or KCL
+    logo: Option<String>,
+    /// A short label message for an exception. e.g. error, warning, help.
+    diag_label: String,
+    /// Code for the current exception type. e.g. E1010, W0091
+    diag_code: Option<String>,
+}
+
+impl LabelPendant {
+    pub fn new(diag_label: String, diag_code: Option<String>) -> Self {
+        Self {
+            logo: None,
+            diag_label,
+            diag_code,
+        }
+    }
+
+    pub fn set_logo(&mut self, logo: String) -> &mut Self {
+        self.logo = Some(logo);
+        self
+    }
+
+    pub fn get_logo(&self) -> String {
```

Comment:
`get_logo` may return a `Option<String>`. Besides, do not use the unwrap() function, use Result and Option whenever possible.